src/app/modules/management/stakeholders/learners/bulk-add/bulk-add.component.ts

```diff
@@ -47,24 +61,46 @@ export class BulkAddComponent implements OnInit {
 
   private createLearner(i: number, elements: string[]): CreateLearner {
     return {
+      id: i + 1,
       num: i + 1,
       index: elements[0].trim(),
       password: elements[1].trim(),
       name: elements[2].trim(),
       surname: elements[3].trim(),
-      email: elements[4].trim()
+      email: elements[4].trim(),
+      userType: this.formGroup.controls['learnersType'].value
     };
   }
 
   onBack(): void {
     this.checkView = false;
+    this.responseView = false;
+    this.responseErrorView = false;
   }
 
   onSubmit(): void {
-    this.dialogRef.close(this.learners);
+    this.learnerService.bulkCreate(this.baseUrl, this.learners).subscribe({
+      next: (data) => {
+        this.responseView = true;
+        this.responseErrorView = false;
+        const accounts = data as BulkAccounts;
+        if (accounts.newAccounts.length !== 0) this.newLearnersAdded = true;
+        this.existingLearners = new MatTableDataSource(accounts.existingAccounts);
+        this.newLearners = new MatTableDataSource(accounts.newAccounts);
+      },
+      error: (error) => {
+        if (error.error.detail.includes('Duplicate username')) this.responseErrorView = true;
```

Comment:
The UI/UX for this is wonky. I would rework the general flow of bulk registration to the following:

1. Admin enters 1 to N accounts.
2. Admin clicks "Proveri".
3. The system displays three lists:
   - "Nalozi koji će biti kreirani" - this list should remove duplicates on the front-end. I would change the current responseErrorView to a snackbar and remove the HTML related to this in bulk-add.component.
   - "Unete stavke koje nemaju dobar format"
   - "Nalozi koji već postoje i neće biti ponovo kreirani"
4. The admin fixes issues or clicks "Save".